src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java

```diff
@@ -281,10 +288,44 @@ public NonBlockingStatsDClient(final String prefix, final int queueSize, String[
                                                           CLIENT_VERSION_TAG + properties.getProperty("dogstatsd_client_version"),
                                                           CLIENT_TAG}, new StringBuilder()).toString();
 
-            this.telemetry = new Telemetry(telemetrytags, statsDProcessor);
+            DatagramChannel telemetryClientChannel = clientChannel;
+            if (addressLookup != telemetryAddressLookup) {
+
+                final SocketAddress telemetryAddress = telemetryAddressLookup.call();
+                if (telemetryAddress instanceof UnixSocketAddress) {
+                    telemetryClientChannel = UnixDatagramChannel.open();
+                    // Set send timeout, to handle the case where the transmission buffer is full
+                    // If no timeout is set, the send becomes blocking
+                    if (timeout > 0) {
+                        telemetryClientChannel.setOption(UnixSocketOptions.SO_SNDTIMEO, timeout);
+                    }
+                    if (bufferSize > 0) {
+                        telemetryClientChannel.setOption(UnixSocketOptions.SO_SNDBUF, bufferSize);
+                    }
+                } else if (transportType == "uds") {
+                    // UDP clientChannel can submit to multiple addresses, we only need
+                    // a new channel if transport type is UDS for main traffic.
+                    telemetryClientChannel = DatagramChannel.open();
+                }
+
+                // similar settings, but a single worker and non-blocking.
+                telemetryStatsDProcessor = createProcessor(queueSize, handler, maxPacketSizeBytes,
+                        poolSize, 1, false);
+            }
 
-            statsDSender = createSender(addressLookup, handler, clientChannel,
-                    statsDProcessor.getBufferPool(), statsDProcessor.getOutboundQueue(), senderWorkers, this.telemetry);
+            this.telemetry = new Telemetry(telemetrytags, telemetryStatsDProcessor);
+
+            statsDSender = createSender(addressLookup, handler, telemetryClientChannel, statsDProcessor.getBufferPool(),
```

Comment:
Is it expected to use `telemetryClientChannel` and not `clientChannel`?

Reply:
I fixed the wrong line 🤦